### PR TITLE
#14: simplified translation and time output

### PR DESCRIPTION
**Pull request by EmilyOng**
with regards to issue #14, i have reduced the complexity of how the markup is being rendered in the translation files by using string formatters, as suggested by @walterbender. i have also made the written/spoken time more direct, eg. one fourty-five PM, midnight, noon, one o'clock AM

### CODE REVIEW

**Review by walterbender — CHANGES_REQUESTED** [review]
A few things seem to have gotten lost along the way:
(1) You need to use gettext for theindividual number and time words, e.g., _("four") on Line 487.
(2) You have lost some of the formatting in your expressions on Lines 499, 501, 503, and 505: %(hours), %(minutes) and %(period),
(3) You can leave some of the markup wrapper outside of the gettext call, e.g., <markup><span lang="en" font_desc="Sans 20">. Just the spans that involve the strings that are translated need to be included inside the gettext call.
(4) re #3 above you need not use gettext for any of the markup on Lines 499 and 501 since there is no special format string used.
(5) You should include a comprehensive TRANS: note explaining what is expected by the translators.
(6) Since we are no longer using time_writer, it should be removed (along with the import and timewriter.py)

**Comment by EmilyOng** [comment]
Thanks for the comments!

- [X] (1) You need to use gettext for the individual number and time words, e.g., _("four") on Line 487.

> I have done that for the individual minutes and hours in the recent commit.

- [X] (2) You have lost some of the formatting in your expressions on Lines 499, 501, 503, and 505: %(hours), %(minutes) and %(period),

> Improved formatting to eg. `%(hour)s % {'hour': ...}`

- [X] (3) You can leave some of the markup wrapper outside of the gettext call, e.g., . Just the spans that involve the strings that are translated need to be included inside the gettext call.

> Left out markup wrapper that targets the entire string outside of gettext and only included relevant markup for colors

- [X] (4) re #3 above you need not use gettext for any of the markup on Lines 499 and 501 since there is no special format string used.]

> Left out all markup outside of gettext and only used formatters within gettext

- [X] (5) You should include a comprehensive TRANS: note explaining what is expected by the translators.

> Included for all gettext references that I have added

- [X] (6) Since we are no longer using time_writer, it should be removed (along with the import and timewriter.py)

> Removed all instances of timewriter and the relevant files

File: clock.py [hunk 1]
```diff
@@ -499,12 +499,12 @@ def _do_write_time(self):
         if hour == 12 and minute == 0:
 
             #TRANS: "Noon", as in afternoon 12PM
-            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%s') + '</span></span></markup>') % ("noon")
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "noon"}
```
Comment (walterbender) [inline clock.py]: This could just be _('noon') instead of _(%('hour')s) % {'hour': _('noon')}

File: clock.py [hunk 2]
```diff
@@ -499,12 +499,12 @@ def _do_write_time(self):
...
         
         elif hour == 0 and minute == 0:
 
             #TRANS: "Midnight", as in morning 12AM
-            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%s') + '</span></span></markup>') % ("midnight")
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "midnight"}
```
Comment (walterbender) [inline clock.py]: same here: just use _('midnight')

File: clock.py [hunk 3]
```diff
@@ -477,14 +475,42 @@ def _write_and_speak(self, speak):
...
+
+        #TRANS: Minute, referring to the unit of time. This list iterates through 60 minutes in word form.
+        minutes = [_('o clock'), _('one'), _('two'), _('three'), _('four'), _('five'), _('six'), _('seven'), _('eight'), _('nine'), _('ten'), _('eleven'), _('twelve'), _('thirteen'), _('fourteen'), _('fifteen'), _('sixteen'), _('seventeen'), _('eighteen'), _('nineteen'), _('twenty'), _('twenty-one'), _('twenty-two'), _('twenty-three'), _('twenty-four'), _('twenty-five'), _('twenty-six'),  _('twenty-seven'), _('twenty-eight'), _('twenty-nine'), _('thirty'), _('thirty-one'), _('thirty-two'), _('thirty-three'), _('thirty-four'), _('thirty-five'), _('thirty-six'), _('thirty-seven'),_('thirty-eight'), _('thirty-nine'), _('fourty'), _('fourty-one'), _('fourty-two'), _('fourty-three'), _('fourty-four'), _('fourty-five'), _('fourty-six'), _('fourty-seven'), _('fourty-eight'), _('fourty-nine'), _('fifty'), _('fifty-one'), _('fifty-two'), _('fifty-three'), _('fifty-four'), _('fifty-five'), _('fifty-six'), _('fifty-seven'), _('fifty-eight'), _('fifty-nine'), _('sixty')]
```
Comment (walterbender) [inline clock.py]: Please make 'o clock' separate from this list with its own TRANS note.

File: clock.py [hunk 4]
```diff
@@ -477,14 +475,42 @@ def _write_and_speak(self, speak):
...
+        #TRANS: AM or PM, referring to morning (before 12PM) or afternoon (12PM and after).
+        period = [_('AM'), _('PM')]
+
+        if hour < 12:
+            periodHour = 0
+        else:
+            periodHour = 1
+
+        if hour == 12 and minute == 0:
+
```
Comment (walterbender) [inline clock.py]: No need for a blank line here

File: clock.py [hunk 5]
```diff
@@ -477,14 +475,42 @@ def _write_and_speak(self, speak):
...
+            #TRANS: "Noon", as in afternoon 12PM
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "noon"}
+
```
Comment (walterbender) [inline clock.py]: No need for a blank line here

File: clock.py [hunk 6]
```diff
@@ -477,14 +475,42 @@ def _write_and_speak(self, speak):
...
+        else:
+            periodHour = 1
+
+        if hour == 12 and minute == 0:
+
+            #TRANS: "Noon", as in afternoon 12PM
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "noon"}
+        
+        elif hour == 0 and minute == 0:
+
```
Comment (walterbender) [inline clock.py]: No need for a blank line here

File: clock.py [hunk 7]
```diff
@@ -477,14 +475,42 @@ def _write_and_speak(self, speak):
...
+            #TRANS: "Midnight", as in morning 12AM
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "midnight"}
+
```
Comment (walterbender) [inline clock.py]: No need for a blank line here

File: clock.py [hunk 8]
```diff
@@ -477,14 +475,42 @@ def _write_and_speak(self, speak):
...
+            #TRANS: "Noon", as in afternoon 12PM
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "noon"}
+        
+        elif hour == 0 and minute == 0:
+
+            #TRANS: "Midnight", as in morning 12AM
+            self._time_in_letters = ('<markup><span lang="en" font_desc="Sans 20"><span foreground="#005FE4">' + _('%(hour)s') + '</span></span></markup>') % {'hour': "midnight"}
+        
+        else:
+
```
Comment (walterbender) [inline clock.py]: No need for a blank line here

**Review by walterbender — CHANGES_REQUESTED** [review]
Nice job addressing my concerns. Just a few minor tweaks left.

**Comment by walterbender** [comment]
One last thing: it would be good to avoid lines longer than 80 characters throughout.

**Comment by leonardcj** [comment]
@EmilyOng @walterbender  Looking forward to a much more easily localized Clock, thanks so much for the improvements in internationalization (i18n).  It is so important for developers to learn about making their code easier to translate (L10n) into many languages by developing their i18n strategy along with the code.

**Pull request merged**